Ponca/src/Fitting/leastSquareLine.h

```diff
@@ -0,0 +1,100 @@
+/*
+ This Source Code Form is subject to the terms of the Mozilla Public
+ License, v. 2.0. If a copy of the MPL was not distributed with this
+ file, You can obtain one at http://mozilla.org/MPL/2.0/.
+*/
+
+#pragma once
+#include "./defines.h"
+
+#include <Eigen/Dense>
+
+namespace Ponca
+{
+
+/*!
+ * \brief Orthogonal Regression is the process of finding the line that best fits a set of points by minimizing their squared orthogonal distances to it.
+    The first key insight is that the orthogonal regression line will always pass through the average (or centroid) of the points.
+    The centroid can be computed by adding the points together, and then dividing by the number of points.
+    Note: Now that we have a point that the line passes through, we just need to calculate its direction.
+
+    Since this algorithm is iterative, we need an iteration function that gets us closer to the true line direction at each step.
```

Comment:
The algorithm you use is not iterative, but based on the Eigen decomposition of the covariance matrix of the points coordinates.